ext/js/background/backend.js

```diff
@@ -2364,19 +2377,14 @@ export class Backend {
         if (typeof deinflect !== 'boolean') { deinflect = true; }
         const enabledDictionaryMap = this._getTranslatorEnabledDictionaryMap(options);
         const {
-            general: {mainDictionary, sortFrequencyDictionary, sortFrequencyDictionaryOrder},
+            general: {mainDictionary, sortFrequencyDictionary, sortFrequencyDictionaryOrder, language},
             scanning: {alphanumeric},
             translation: {
-                convertHalfWidthCharacters,
-                convertNumericCharacters,
-                convertAlphabeticCharacters,
-                convertHiraganaToKatakana,
-                convertKatakanaToHiragana,
-                collapseEmphaticSequences,
                 textReplacements: textReplacementsOptions,
                 searchResolution
             }
         } = options;
+        const textTransformationsOptions = options.languages[language].textTransformations || {};
```

Comment:
`?.textTransformations`

Values can be omitted based on the JSON spec I believe. Typescript doesn't catch this situation unfortunately, I suppose it's similar how it doesn't (or can't) do bounds checks for array types.

Reply:
(This recommendation may change if I think think we should switch to using arrays instead of object dictionaries. Haven't gotten back to that yet.)